Allow APPLICATION_PATH to be defined in Apache as an environment vari…

EreMaijala opened the pull request:
…able so that __DIR__ is not used (doesn't work properly with symlinked files). Also use APPLICATION_PATH in theme base directory.

At least the index.php change is needed for us to be able to use a mostly symlinked (apart from a custom theme and maybe a module) VuFind installation for all institutions. The theme change would be nice so we wouldn't need to override it locally.

demiankatz commented:
This seems reasonable to me, though I'm not sure if the long line in public/index.php is going to cause a style complaint (right now, Travis seems to be unhappy).

It's also probably worth adding the new environment variable to the default config/vufind/httpd-vufind.conf for easy reference. (Not sure if it's best to have it commented out by default or to actually fill it in with install.php -- but either way, we should add *something* there).

EreMaijala commented:
Added an example in httpd-vufind.conf and fixed the long lines (there were some already before). The Travis failure seems to have been a temporary failure setting up the build.

demiankatz commented:
Thanks again! Manually merged to master.

I wonder if we might want to do something better than using APPLICATION_PATH in VuFindTheme, since it was intended that that module might eventually become something reusable, and the hard-coded dependency on a constant would be a problem outside the context of the VuFind application... but we can worry about that later, if we actually need to split out the component.
